documentation/source/user_section/installation/linux.rst

```diff
@@ -214,14 +192,32 @@ First, save your Docker image:
 
       docker commit <CONTAINER_ID> <YOUR_NAME>/<DISTROS>:<VERSION>
 
-Create an X11 server for handling display:
+Forward X11 display protocol:
+
+.. note::
+
+   The following instructions apply to hosts running the Xorg X11 server.
+
+   Set up may vary if you are using a different X11 server.
+
+1. Install ``xauth`` and ``xhost`` on host machine:
+
+   For example on Debian/Ubuntu:
+
+   .. code:: bash
+
+      apt install xauth x11-xserver-utils
+
+2. Permit docker access to the X11 Server
+
+   If hosting container from the local machine:
+
+   .. code:: bash
+
+      xhost +local:docker
 
-1. Install XQuartz X11 server.
-2. Check ‘Allow connections from network clientsoption inXQuartz\` settings.
-3. Quit and restart XQuartz.
-4. In XQuartz window xhost + 127.0.0.1
-5. In your other Terminal window, run:
-   ``docker run -ti --rm -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix <CONTAINER_ID>``
+3. In your Terminal window, run:
+   ``docker run -it --rm --privileged -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix <CONTAINER_ID>``
```

Comment:
Hmm. After I got the `docker run` command working, I tried running `sct_check_dependencies` to see if PyQt would work. But, I got the following error:

```
Check if figure can be opened with PyQt.............qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb.
```

I then tried again with `export QT_DEBUG_PLUGINS=1` specified (as you suggested), and got the following output:

<details>

```
Check if figure can be opened with PyQt.............QFactoryLoader::QFactoryLoader() checking directory path "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms" ...
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqeglfs.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqeglfs.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "eglfs"
        ]
    },
    "archreq": 0,
    "className": "QEglFSIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("eglfs")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqlinuxfb.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqlinuxfb.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "linuxfb"
        ]
    },
    "archreq": 0,
    "className": "QLinuxFbIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("linuxfb")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqminimal.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqminimal.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "minimal"
        ]
    },
    "archreq": 0,
    "className": "QMinimalIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("minimal")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqminimalegl.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqminimalegl.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "minimalegl"
        ]
    },
    "archreq": 0,
    "className": "QMinimalEglIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("minimalegl")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqoffscreen.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqoffscreen.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "offscreen"
        ]
    },
    "archreq": 0,
    "className": "QOffscreenIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("offscreen")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqvnc.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqvnc.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "vnc"
        ]
    },
    "archreq": 0,
    "className": "QVncIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("vnc")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-egl.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-egl.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "wayland-egl"
        ]
    },
    "archreq": 0,
    "className": "QWaylandEglPlatformIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("wayland-egl")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-generic.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-generic.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "wayland"
        ]
    },
    "archreq": 0,
    "className": "QWaylandIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("wayland")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-xcomposite-egl.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-xcomposite-egl.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "wayland-xcomposite-egl"
        ]
    },
    "archreq": 0,
    "className": "QWaylandXCompositeEglPlatformIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("wayland-xcomposite-egl")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-xcomposite-glx.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwayland-xcomposite-glx.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "wayland-xcomposite-glx"
        ]
    },
    "archreq": 0,
    "className": "QWaylandXCompositeGlxPlatformIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("wayland-xcomposite-glx")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwebgl.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqwebgl.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "webgl"
        ]
    },
    "archreq": 0,
    "className": "QWebGLIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("webgl")
QFactoryLoader::QFactoryLoader() looking at "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqxcb.so"
Found metadata in lib /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqxcb.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "xcb"
        ]
    },
    "archreq": 0,
    "className": "QXcbIntegrationPlugin",
    "debug": false,
    "version": 330752
}


Got keys from plugin meta data ("xcb")
QFactoryLoader::QFactoryLoader() checking directory path "/sct/python/envs/venv_sct/bin/platforms" ...
Cannot load library /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqxcb.so: (libXrender.so.1: cannot open shared object file: No such file or directory)
QLibraryPrivate::loadPlugin failed on "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqxcb.so" : "Cannot load library /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqxcb.so: (libXrender.so.1: cannot open shared object file: No such file or directory)"
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb.
```
</details>

The key line is:

```
Cannot load library /sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqxcb.so: (libXrender.so.1: cannot open shared object file: No such file or directory)
```

To fix this, I ran the following command inside the container:

```
apt install libxrender1
```

I then re-ran `sct_check_dependencies` but got the following new error:

```
QFactoryLoader::QFactoryLoader() checking directory path "/sct/python/envs/venv_sct/bin/platforms" ...
loaded library "/sct/python/envs/venv_sct/lib/python3.9/site-packages/PyQt5/Qt/plugins/platforms/libqxcb.so"
qt.qpa.xcb: could not connect to display :0.0
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```

I am stuck here. I have tried the following solutions:

- Running [`xhost +local:docker`](https://stackoverflow.com/questions/28392949/running-chromium-inside-docker-gtk-cannot-open-display-0#comment57024354_28395350)
- Running [`xhost local:root`](https://stackoverflow.com/a/34586732)
- Adding [`:rw`](https://stackoverflow.com/q/69872788) to the `-v` option.
- Adding [`--net=host`](https://unix.stackexchange.com/a/707724/583446) to the `docker run` command.
- Installing [additional dependencies](https://stackoverflow.com/a/76610935).

But all of this results in the same output. If you have any ideas, please let me know. :slightly_smiling_face:

Reply:
Thanks for your prompt and thorough reply @joshuacwnewton .
I have been testing on Arch Linux with the Docker CLI, and it seems as though there are differences between Docker Desktop and the Docker CLI.
- libxrender was not needed on my Arch machine, but I found that it was necessary for my Windows machine (Docker Desktop).
- the /tmp/.X11-unix directory was permitted as shared by default on my Arch machine, I found that I didn't need to mount the X11 Unix socket on Windows.

It seems like it would be something to do with the X11 forwarding through Docker Desktop. I am currently installing Docker Desktop on my Arch machine and will see if I can find anything.

Reply:
Ahhh, thank you so much for the investigation, and for examining the various difference between OSs and versions of Docker. :hearts: